java/core/src/test/java/com/whylogs/api/resultsets/TestProfileResultSet.java

```diff
@@ -0,0 +1,79 @@
+package com.whylogs.api.resultsets;
+
+import com.whylogs.api.logger.resultSets.ProfileResultSet;
+import com.whylogs.core.DatasetProfile;
+import com.whylogs.core.errors.Error;
+import com.whylogs.core.metrics.IntegralMetric;
+import com.whylogs.core.metrics.MetricConfig;
+import com.whylogs.core.schemas.DatasetSchema;
+import java.util.HashMap;
+import java.util.InputMismatchException;
+import java.util.Optional;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+@Test
+public class TestProfileResultSet {
+
+  private ProfileResultSet defaultResultSet() {
+    HashMap<String, Object> data = new HashMap<>();
+    data.put("test", 1);
+    data.put("test2", "2");
+    DatasetSchema datasetSchema = new DatasetSchema();
+    datasetSchema.resolve(data);
+
+    DatasetProfile datasetProfile =
+        new DatasetProfile(Optional.of(datasetSchema), Optional.empty(), Optional.empty());
```

Comment:
Consider: Even if just for our internal use, it may be worth a ctor overload that takes in 0 or 1 parameters?

Reply:
Yeah I was thinking the same thing. I even have a note on it. I don't like all these optionals as the main workflow.  I was planning on doing it in the next iteration.